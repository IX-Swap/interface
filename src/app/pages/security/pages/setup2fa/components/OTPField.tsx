import React from 'react'
import { TypedField } from 'components/form/TypedField'
import { Input } from '@mui/material'
import { useFormContext } from 'react-hook-form'
import { Enable2faFormValues } from 'app/pages/security/pages/setup2fa/types'

export const OTPField = () => {
  const { control } = useFormContext<Enable2faFormValues>()

  return (
    <TypedField
      fullWidth
      control={control}
      component={Input}
      label='OTP'
      name='otp'
      autoComplete='off'
    />
  )
}
