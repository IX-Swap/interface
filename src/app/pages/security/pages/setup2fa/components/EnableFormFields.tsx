import React from 'react'
import { TypedField } from 'components/form/TypedField'
import { OTPField } from 'components/form/OTPField'
import { useFormContext } from 'react-hook-form'
import { plainValueExtractor } from 'helpers/forms'

export const EnableFormFields = () => {
  const { control } = useFormContext()

  return (
    <TypedField
      control={control}
      customRenderer
      component={OTPField}
      name='otp'
      label=''
      variant='standard'
      valueExtractor={plainValueExtractor}
      shouldAutoFocus
    />
  )
}
