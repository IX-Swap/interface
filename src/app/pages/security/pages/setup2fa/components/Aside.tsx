import React from 'react'
import { Box, Typography } from '@material-ui/core'
import { AppRouterLink } from 'components/AppRouterLink'
import { VSpacer } from 'components/VSpacer'
import { useStyles } from 'app/pages/security/pages/setup2fa/Setup2fa.styles'
import { SecurityRoute } from 'app/pages/security/router/config'

export const Aside = () => {
  const { aside } = useStyles()

  return (
    <Box className={aside}>
      <Typography variant='h5'>What is 2FA?</Typography>
      <VSpacer size='medium' />
      <Typography variant='body1'>
        (2FA) Two-Factor Authentication increases your account security by
        requiring multiple forms of verification when signing into InvestaX
        Prime.
      </Typography>
      <VSpacer size='medium' />
      <AppRouterLink variant='body1' to={SecurityRoute.guide}>
        Learn more
      </AppRouterLink>
    </Box>
  )
}
