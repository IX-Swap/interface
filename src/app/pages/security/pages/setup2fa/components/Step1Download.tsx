import React from 'react'
import { Typography, Grid, Link, Box } from '@material-ui/core'
import appStoreLogo from '../assets/appstore.png'
import googlePlayLogo from '../assets/googleplay.png'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step1Download = () => (
  <StepWrapper title='Install Authenticator App in Your Smartphone Device'>
    <Grid container direction='column' spacing={3} alignItems='center'>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          When you log in, you’ll need to enter a code generated by the Google
          Authenticator app or another authenitcator app
        </Typography>
      </Grid>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          Download an authentication app to your smartphone from the App Store
          (for iPhone) or Google Play (for android smartphones)
        </Typography>
      </Grid>
      <Grid item>
        <Box paddingY={4}>
          <Grid container justify='center' spacing={2}>
            <Grid item>
              <Link
                href='https://apps.apple.com/us/app/google-authenticator/id388497605'
                target='_blank'
              >
                <img height='40px' src={appStoreLogo} alt='App Store Logo' />
              </Link>
            </Grid>
            <Grid item>
              <Link
                href='https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en'
                target='_blank'
              >
                <img
                  height='40px'
                  src={googlePlayLogo}
                  alt='Google Play Logo'
                />
              </Link>
            </Grid>
          </Grid>
        </Box>
      </Grid>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          Please click “NEXT” when you have finished the authenticator app
          installed in your smartphone
        </Typography>
      </Grid>
    </Grid>
  </StepWrapper>
)
