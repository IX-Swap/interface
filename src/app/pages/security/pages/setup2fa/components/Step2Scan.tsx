import React from 'react'
import { Typography, Box, Grid, Link } from '@material-ui/core'
import useStyles from './Step2Scan.styles'
import { useSetup2fa } from '../hooks/useSetup2fa'
import { useSetup2faStore } from '../context'
import { useObserver } from 'mobx-react'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step2Scan = () => {
  const classes = useStyles()
  const store = useSetup2faStore()
  const { isLoading } = useSetup2fa()

  return useObserver(() => (
    <StepWrapper title='Scan This QR Code in Your Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Use the app to scan the barcode below or type in the key. If you
            don’t have a smartphone or need more help, please{' '}
            <Link href='https://investax.io/contact/'>contact us.</Link>
          </Typography>
        </Grid>
        <Grid item>
          {!isLoading && (
            <Grid container justifyContent='center'>
              <Box pt={4} pb={3}>
                <Grid container justifyContent='center' alignItems='center'>
                  <Grid item>
                    <div
                      data-testid='store-image'
                      className={classes.image}
                      style={{
                        backgroundImage: `url('${store.image}')`,
                        marginBottom: 10
                      }}
                    />
                  </Grid>
                  <Grid item>
                    <Typography className={classes.label}>
                      Click the “add icon “ on the app to scan the barcode below
                      or type in the key.
                    </Typography>
                    <Typography variant='h5' className={classes.key}>
                      {store.key}
                    </Typography>
                  </Grid>
                </Grid>
              </Box>
            </Grid>
          )}
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please click “NEXT” when you have sucessfully scanned...
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  ))
}
