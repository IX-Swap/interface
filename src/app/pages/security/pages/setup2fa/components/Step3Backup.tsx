import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import useStyles from './Step3Backup.styles'
import { useSetup2faStore } from '../context'
import { useObserver } from 'mobx-react'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step3Backup = () => {
  const classes = useStyles()
  const store = useSetup2faStore()

  return useObserver(() => (
    <StepWrapper title='Save This Backup Key'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please save this key code in a safe place. It will allow you to
            recover your authenticator in case of losing your phone
          </Typography>
        </Grid>
        <Grid item>
          <Typography variant='h4' align='center' className={classes.key}>
            {store.key}
          </Typography>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please click “Next” if you have sucessfully stored this key.
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  ))
}
