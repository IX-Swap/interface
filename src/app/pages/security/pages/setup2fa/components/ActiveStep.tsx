import React from 'react'
import { Step2Scan } from 'app/pages/security/pages/setup2fa/components/Step2Scan/Step2Scan'
import { Step3Backup } from 'app/pages/security/pages/setup2fa/components/Step3Backup/Step3Backup'
import { Step4Enable } from 'app/pages/security/components/Step4Enable/Step4Enable'
import { Enabled } from 'app/pages/security/pages/setup2fa/components/Enabled'
import { TwoFaData } from 'app/pages/security/types'
import { Step1Download } from 'app/pages/security/pages/setup2fa/components/Step1Download/Step1Download'

export interface ActiveStepProps {
  index: number
  twoFaData: TwoFaData | undefined
  nextStep: () => void
}

export const ActiveStep = ({ index, twoFaData, nextStep }: ActiveStepProps) => {
  switch (index) {
    case 0:
      return <Step1Download />
    case 1:
      return <Step2Scan twoFaData={twoFaData} />
    case 2:
      return <Step3Backup twoFaData={twoFaData} />
    case 3:
      return <Step4Enable nextStep={nextStep} />
    default:
      return <Enabled />
  }
}
