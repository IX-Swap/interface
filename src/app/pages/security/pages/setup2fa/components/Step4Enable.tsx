import React from 'react'
import { Typography, Box, Grid } from '@mui/material'
import { useEnable2fa } from '../hooks/useEnable2fa'
import { Enable2faFormValues } from '../types'
import { enable2faFormValuesSchema } from '../validation'
import { Form } from 'components/form/Form'
import { Submit } from 'components/form/Submit'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/pages/setup2fa/components/EnableFormFields'

const defaultValues = {
  otp: ''
}

export const Step4Enable = () => {
  const [enable2fa] = useEnable2fa()
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper title='Enable Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please enter the 6 digit code from your authenticator app
          </Typography>
        </Grid>
        <Grid item>
          <Form
            defaultValues={defaultValues}
            validationSchema={enable2faFormValuesSchema}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <EnableFormFields />
            <Box my={4} width='100%' textAlign='center'>
              <Submit>Enable</Submit>
            </Box>
          </Form>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Once enabled your authenticator you will be logged out and re-enter
            the OTP in order to access the platform
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
