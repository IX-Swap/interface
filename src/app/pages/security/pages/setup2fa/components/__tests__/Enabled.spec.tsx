import React from 'react'
import { render } from 'test-utils'
import { Enabled } from 'app/pages/security/pages/setup2fa/components/Enabled'

describe('Enabled', () => {
  it('should match snapshot', () => {
    const { container } = render(<Enabled />)
    expect(container).toMatchSnapshot()
  })
})
