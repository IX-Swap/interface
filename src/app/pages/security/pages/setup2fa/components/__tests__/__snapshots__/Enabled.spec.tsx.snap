// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enabled should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-11ypfzm-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1cea49u-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        style="color: rgb(76, 175, 80); width: 120px; height: 120px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <div
        class="MuiBox-root css-1jpm9pd"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1f4u6as-MuiTypography-root"
      >
        2FA Enabled
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6 css-1er4735-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-6bx0l6-MuiTypography-root"
      >
        Your account is now protected by Two-Factor Authentication
      </p>
    </div>
  </div>
</div>
`;

exports[`Enabled should match snapshot, invokes history push on button click 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-30jy6e-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1o62c1b-MuiGrid-root"
    >
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-118amwv-MuiTypography-root"
        >
          2-FA Has Been Updated Successfully.
        </h4>
      </div>
      <div
        class="MuiGrid-root css-186c2wj-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-text-1 css-6bx0l6-MuiTypography-root"
      >
        Your account is now protected by Two-Factor Authentication (2-FA).
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-7kz4db-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1i8ofwk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Ok
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
